encoding.go

```diff
@@ -131,6 +132,14 @@ func newBuffer(data []byte) *byteBuffer {
 	}
 }
 
+func newBufferFromInt(num uint64) *byteBuffer {
+	data := make([]byte, 8)
+	binary.BigEndian.PutUint64(data, num)
+	return &byteBuffer{
```

Comment:
Minor: I prefer we `return newBuffer(bytes.TrimLeft(data, "\x00"))`, but it's up to you.